packages/kbn-zod-helpers/src/non_empty_string.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import * as z from '@kbn/zod';
+export function isNonEmptyString(input: string, ctx: z.RefinementCtx) {
+  if (typeof input === 'string' && input.trim() === '') {
```

Comment:
nit: the `typeof input === 'string'` check should be redundant since this function is called after zod validates that the input is a string